lib/Service/MailManager.php

```diff
@@ -585,6 +585,40 @@ public function deleteMailbox(Account $account,
 		$this->mailboxMapper->delete($mailbox);
 	}
 
+	/**
+	 * Clear messages in folder
+	 *
+	 * @param Account $account
+	 * @param Mailbox $mailbox
+	 *
+	 * @throws DoesNotExistException
+	 * @throws Horde_Imap_Client_Exception
+	 * @throws Horde_Imap_Client_Exception_NoSupportExtension
+	 * @throws ServiceException
+	 */
+	public function clearMailbox(Account $account,
+								  Mailbox $mailbox): void {
+		$client = $this->imapClientFactory->getClient($account);
+		$trashMailboxId = $account->getMailAccount()->getTrashMailboxId();
```

Comment:
``$trashMailboxId`` can be `null`

Reply:
So if null, then just delete permanently, right?

Reply:
Yes :)

Reply:
This method is deleted